homeassistant/components/trafikverket_train/sensor.py

```diff
@@ -0,0 +1,147 @@
+"""
+Train information for departures and delays, provided by Trafikverket.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.trafikverket_train/
+"""
+
+from datetime import date, datetime, timedelta
+import asyncio
+import logging
+import voluptuous as vol
+
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_API_KEY, CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.aiohttp_client import (async_get_clientsession)
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TRAINS = "trains"
+CONF_FROM = "from"
+CONF_TO = "to"
+CONF_TIME = "time"
+
+ATTR_CANCELED = "canceled"
+ATTR_DELAY_TIME = "number_of_minutes_delayed"
+ATTR_PLANNED_TIME = "planned_time"
+ATTR_ESTIMATED_TIME = "estimated_time"
+ATTR_ACTUAL_TIME = "actual_time"
+ATTR_OTHER_INFORMATION = "other_information"
+ATTR_DEVIATIONS = "deviations"
+
+ICON = "mdi:train"
+SCAN_INTERVAL = timedelta(minutes=10)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_TRAINS): [{
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_TO): cv.string,
+        vol.Required(CONF_FROM): cv.string,
+        vol.Optional(CONF_TIME): cv.time}]
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(
+        hass, config, async_add_devices, discovery_info=None):
+    """Set up the departure sensor."""
+    _LOGGER.debug("Start async_setup_platform")
+    from pytrafikverket import TrafikverketTrain
+    httpsession = async_get_clientsession(hass)
+    train_api = TrafikverketTrain(httpsession, config.get(CONF_API_KEY))
+    sensors = []
+    for train in config.get(CONF_TRAINS):
+        from_sig = yield from train_api.async_get_train_station(
+            train.get(CONF_FROM))
+        to_sig = yield from train_api.async_get_train_station(
+            train.get(CONF_TO))
+        sensor = TrainSensor(train_api,
+                             train.get(CONF_NAME),
+                             from_sig,
+                             to_sig,
+                             train.get(CONF_TIME))
+        sensors.append(sensor)
+
+    async_add_devices(sensors, update_before_add=True)
+    _LOGGER.debug("end async_setup_platform")
```

Comment:
Would be more helpful if the message would say that connection X was set up and connection Y in the loop.

Reply:
Removed it, was leftover from development debug logging.